src/OData.QueryBuilder/Extensions/StringExtensions.cs

```diff
@@ -35,14 +35,27 @@ public static IEnumerable<string> ReplaceWithStringBuilder(this IEnumerable<stri
             return replaceValues;
         }
 
+        public static string replaceSpecialCharacters(string attribute)
+        {
+            attribute = attribute.Replace("'", "''");
+            attribute = attribute.Replace("+", "%2B");
+            attribute = attribute.Replace("/", "%2F");
+            attribute = attribute.Replace("?", "%3F");
+            attribute = attribute.Replace("%", "%25");
+            attribute = attribute.Replace("#", "%23");
+            attribute = attribute.Replace("&", "%26");
+
+            return attribute;
+        }
+        
         public static string ToQuery(this object @object)
         {
             switch (@object)
             {
                 case null:
                     return "null";
                 case string @string:
-                    return $"'{@string}'";
+                    return $"'{replaceSpecialCharacters(@string)}'";
```

Comment:
Hi! Try using the method, it seems to do the same
https://github.com/ZEXSM/OData.QueryBuilder#replacecharacters